packages/expect/src/index.ts

```diff
@@ -363,19 +362,16 @@ const makeThrowingMatcher = (
             })();
 
       if (isPromise(potentialResult)) {
-        const asyncResult = potentialResult as AsyncExpectationResult;
```

Comment:
only change to non-test code. The cast was unnecessary as TS could infer it from it being a promise